test/integration_test.go

```diff
@@ -1951,6 +1955,46 @@ func TestDependencyOutputRegression854(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// Regression testing for https://github.com/gruntwork-io/terragrunt/issues/906
+func TestDependencyOutputRegression906(t *testing.T) {
```

Comment:
Minor nit: would probably be better if the function name reflected what was tested (e.g., TestDependencyOutputConcurrency), as the issue link is already in the comment.

Reply:
Done in 8362023